lib/github_service/commands/run_test.rb

```diff
@@ -0,0 +1,302 @@
+    # == Examples
+    #
+    # === In ManageIQ/manageiq PR #1234:
+    #
+    # The following command:
+    #
+    #   @miq-bot run-tests manageiq-ui-classic
+    #
+    # will create a commit with the .travis.yml changes:
+    #
+    #
+    #   @@ -14,6 +14,6 @@ matrix:
+    #      fast_finish: true
+    #    env:
+    #      global:
+    #   -  - REPOS=
+    #   +  - REPOS=ManageIQ/manageiq#12345
+    #      matrix:
+    #   -  - TEST_REPO=
+    #   +  - TEST_REPO=ManageIQ/manageiq-ui-classic
+    #
+    #
+    # === In ManageIQ/manageiq PR #1234
+    #
+    # The following command:
+    #
+    #   @miq-bot run-tests manageiq-api,manageiq-ui-classic \
+    #     including Fryguy/more_core_extensions@feature,Fryguy/linux_admin@feature
+    #
+    # will create a commit with the .travis.yml changes:
+    #
+    #   @@ -14,6 +14,6 @@ matrix:
+    #      fast_finish: true
+    #    env:
+    #      global:
+    #   -  - REPOS=
+    #   +  - REPOS=Fryguy/more_core_extensions@feature,Fryguy/linux_admin@feature,ManageIQ/manageiq#1234
+    #      matrix:
+    #   -  - TEST_REPO=
+    #   +  - TEST_REPO=ManageIQ/manageiq-api
+    #   +  - TEST_REPO=ManageIQ/manageiq-ui-classic
+    #
+    # TODO:  Handle the "self" case, where `manageiq` is also a TEST_REPO
+    #
+    # (maybe include a "self" helper as well?)
+    #
+    # === In ManageIQ/manageiq-ui-classic PR #1234:
+    #
+    # The following command:
+    #
+    #   @miq-bot run-tests manageiq#5678
+    #
+    # will create a commit with the .travis.yml changes:
+    #
+    #   @@ -14,6 +14,6 @@ matrix:
+    #      fast_finish: true
+    #    env:
+    #      global:
+    #   -  - REPOS=
+    #   +  - REPOS=manageiq#5678
+    #      matrix:
+    #   -  - TEST_REPO=
+    #   +  - TEST_REPO=manageiq-ui-classic#1234
+    #
+    class RunTest < Base
+      # The user calling the command
+      attr_reader :issuer
+
+      # The org of the issue this command originated from
+      attr_reader :org
+
+      # The (extra) repo(s) being targeted to be included in the test run
+      attr_reader :repos
+
+      # The repo(s) that will have the test suite run
+      attr_reader :test_repos
+
+      # The *-cross_repo-tests rugged instance
+      attr_reader :rugged_repo
+
+      # The arguments for the `run-test` command being called
+      attr_reader :value
+
+      restrict_to :organization
+
+      def self.test_repo_url
+        Settings.github.run_tests_repo.url
+      end
+
+      def self.test_repo_name
+        Settings.github.run_tests_repo.name
+      end
+
+      def self.test_repo_clone_dir
+        @test_repo_clone_dir ||= begin
+                                   url_parts = test_repo_url.split("/")[-2, 2]
+                                   repo_org  = url_parts.first
+                                   repo_dir  = test_repo_name
+                                   File.join(::Repo::BASE_PATH, repo_org, repo_dir)
+                                 end
+      end
+
+      # Name of the bot
```

Comment:
Going to go out on a limb and say that this method name is self documenting :) A few of these comments are probably not necessary

Reply:
Well, I was going to say I was just being consistent, but I guess I was only consistent when defining the `attr_readers`.  I probably am find removing this.